src/dbus/server/dbus_thread_object.cpp

```diff
@@ -1522,6 +1524,22 @@ void DBusThreadObject::ActiveDatasetChangeHandler(const otOperationalDatasetTlvs
     SignalPropertyChanged(OTBR_DBUS_THREAD_INTERFACE, OTBR_DBUS_PROPERTY_ACTIVE_DATASET_TLVS, value);
 }
 
+void DBusThreadObject::LeaveNetworkHandler(DBusRequest &aRequest)
+{
+    mNcp->GetThreadHelper()->DetachGracefully([aRequest, this](otError error) mutable {
+        SuccessOrExit(error);
+        SuccessOrExit(error = otInstanceErasePersistentInfo(mNcp->GetThreadHelper()->GetInstance()));
```

Comment:
We are saving old local on-link prefixes in settings, see https://github.com/openthread/openthread/pull/8330.

The change here will erase this information when leave current network, but the old local on-link prefixes should persist between network migrations. 

What about erasing only Thread operational datasets? @bukepo @abtink

Reply:
I think if we fully leave the Thread network we should erase everything. 
- We don't want to reuse any of the previous info associated with old network (e.g. SRP key, child table, parent info, mac address

If we migrate from one network to another, then may be fine to keep some of the old info.